apps/web/src/server/app/app.express.js

```diff
@@ -53,27 +54,45 @@ app.use(bodyParser.json());
 // Parse Cookie header and populate req.cookies with an object keyed by the cookie names.
 app.use(cookieParser());
 
+// Generate the nonce for each request.  The exception is in test to allow for snapshots.
+app.use((req, res, next) => {
+	if (!config.isTest) {
+		res.locals.cspNonce = crypto.randomBytes(16).toString('hex');
+	}
+	next();
+});
+
+/**
+ * Middleware to return current nonce
+ *
+ * @type {import('express').RequestHandler}
+ * @returns {string}
+ */
+const addCSPNonce = (req, res) => `'nonce-${res.locals.cspNonce}'`;
+
 // Secure apps by setting various HTTP headers
-app.use(helmet());
 app.use(
-	helmet.contentSecurityPolicy({
-		directives: {
-			scriptSrc: ["'self'", () => `'nonce-${locals.cspNonce}'`],
-			defaultSrc: ["'self'", config.blobStorageUrl],
-			'font-src': ["'self'"],
-			'img-src': ["'self'", config.blobStorageUrl],
-			'style-src': [
-				"'self'",
-				// the hashes below are for the toastui editor, used for rich-text editing
-				// allowing specific styles by hash is still safe enough
-				"'unsafe-hashes'",
-				"'sha256-ZdHxw9eWtnxUb3mk6tBS+gIiVUPE3pGM470keHPDFlE='",
-				"'sha256-dCNOmK/nSY+12vHzasLXiswzlGT5UHA7jAGYkvmCuQs='",
-				"'sha256-aqNNdDLnnrDOnTNdkJpYlAxKVJtLt9CtFLklmInuUAE='",
-				// these are for the html-content-editor warning message show/hide logic
-				"'sha256-dCNOmK/nSY+12vHzasLXiswzlGT5UHA7jAGYkvmCuQs='",
-				"'sha256-ZdHxw9eWtnxUb3mk6tBS+gIiVUPE3pGM470keHPDFlE='"
-			]
+	helmet({
+		contentSecurityPolicy: {
+			directives: {
+				// @ts-ignore
+				scriptSrc: ["'self'", "'unsafe-eval'", addCSPNonce],
```

Comment:
We don't need 'unsafe-eval'

Reply:
Removed as requested, retested and still works